### PR TITLE
chore: fixes for legacy amino encoding for ledger signing

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #4104

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/applications/transfer/v1/tx.proto
Comment (damiannolan) [inline proto/ibc/applications/transfer/v1/tx.proto:1]: They changes here still yield: `ERR failure when running app err="proto: not found"` when signing with ledger

**Comment by damiannolan** [comment]
See SDK issue for more details: https://github.com/cosmos/cosmos-sdk/issues/17975

The changes in this PR are still relevant and should be included. The issue is due to the internal proto registries not being populated with ibc types.

File: proto/ibc/applications/transfer/v1/tx.proto [hunk 1]
```diff
@@ -36,14 +36,14 @@ message MsgTransfer {
   // the channel by which the packet will be sent
   string source_channel = 2;
   // the tokens to be transferred
-  cosmos.base.v1beta1.Coin token = 3 [(gogoproto.nullable) = false, (amino.encoding) = "legacy_coin"];
```
Comment (damiannolan) [inline proto/ibc/applications/transfer/v1/tx.proto:39]: `(amino.encoding) = "legacy_coin"` instructions the amino codec on how to encode `nil` slices of `Coin` - you can see that here -> https://github.com/cosmos/cosmos-sdk/blob/main/x/tx/signing/aminojson/json_marshal.go#L70

**Comment by damiannolan** [comment]
Tested local signing using https://github.com/cosmos/cosmos-sdk/pull/17996

File: proto/ibc/applications/transfer/v1/tx.proto
Comment (damiannolan) [reply proto/ibc/applications/transfer/v1/tx.proto:1]: Fixed by https://github.com/cosmos/cosmos-sdk/pull/17996

**Review by chatton — APPROVED** [review]
LGTM, walked through together. Worth noting it's been tested locally with `MsgTransfer` and `MsgPayPacketFeeAsync` as well as `MsgBankSend`

Thanks for figuring this out one 💪

**Review by srdtrk — APPROVED** [review]
LGTM. Thanks so much

**Review by charleenfei — APPROVED** [review]

**Pull request merged**